js/bybit.js

```diff
@@ -0,0 +1,1001 @@
+        const data = this.safeValue (response, 'result', []);
+        if (data.length > 0) {
+            for (let i = 0; i < data.length; i++) {
+                const order = data[i];
+                const side = this.safeString (order, 'side');
+                let bidsOrAsks = undefined;
+                if (side === 'Sell') {
+                    bidsOrAsks = 'asks';
+                } else if (side === 'Buy') {
+                    bidsOrAsks = 'bids';
+                }
+                const amount = this.safeFloat (order, 'size');
+                const price = this.safeFloat (order, 'price');
+                if (bidsOrAsks !== undefined) {
+                    result[bidsOrAsks].push ([ price, amount ]);
+                }
+            }
+            result['bids'] = this.sortBy (result['bids'], 0, true);
+            result['asks'] = this.sortBy (result['asks'], 0);
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        const response = await this.wapiGetPositionList (this.extend (request, params));
+        const retData = this.safeValue (response, 'result');
+        const result = { 'info': retData };
+        for (let i = 0; i < retData.length; i++) {
+            const position = retData[i];
+            const symbol = this.safeString (position, 'symbol');
+            const market = this.markets_by_id[symbol];
+            const currencyId = market['base'];
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['total'] = position['wallet_balance'];
+            account['used'] = position['position_margin'] + position['occ_closing_fee'] + position['occ_funding_fee'] + position['order_margin'];
+            result[code] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+            'interval': this.timeframes[timeframe],
+        };
+        if (since === undefined) {
+            request['from'] = this.seconds () - 86400; // default from 24 hours ago
+        } else {
+            request['from'] = this.truncate (since / 1000, 0);
+        }
+        if (limit !== undefined) {
+            request['limit'] = limit; // default == max == 200
+        }
+        const response = await this.v2GetPublicKlineList (this.extend (request, params));
+        return this.parseOHLCVs (this.safeValue (response, 'result', {}), market, timeframe, since, limit);
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        // There are two type of orders, so user must define which order to cancel
```

Comment:
This comment looks redundant (it's about canceling, not fetching the orders).
Also, `fetchOrder` should not be emulated with `fetchOrders` in this case. I'd rather remove it entirely, if the exchange does not have an endpoint for fetching a specific order by id.

Reply:
The point is that exchange has 2 different endpoints for orders: active orders (https://github.com/bybit-exchange/bybit-official-api-docs/blob/master/en/rest_api.md#get-active-order) and conditional orders (https://github.com/bybit-exchange/bybit-official-api-docs/blob/master/en/rest_api.md#get-conditional-order). If request includes additional parameter (order_id (active orders endpoint) or stop_order_id (conditional orders enpoint)) then only one order gets returned - the one with requested id. So fetch orders in this function actually will do a request for one order with defined order_id or stop_order_id. I decided it would be correct decision to ask user to provide additional paramets to have a possibility of getting both types of orders.
Another problem in this code implementation is that fetchOrders method fetches only active orders (if order_id not defined in params) because endpoints for active orders and conditional orders require the same parameters (except not necessary order_id and stop_order_id params) so i dont't know how let user get both lists of orders using one method fetchOrders. Any ideas?

Reply:
We don't need the `fetchOrders` in this case, just the `fetchOrder` if it can return an open or closed order by id regardless of the state of the order. 

If the exchange cannot do that, then we don't have `fetchOrders` nor `fetchOrder` and only have `fetchOpenOrders` and `fetchClosedOrders`. In some cases we also add `fetchClosedOrder` and/or `fetchOpenOrder` (for single active/inactive order).

Reply:
From their docs it looks like the exchange has just the `fetchOpenOrders` and `fetchOpenOrder`. Nothing about the historical, or, closed orders. As for the usual orders vs conditional orders – overriding the endpoint via params is ok, also see how the `fetchOrderMethod` or `createOrderMethod` options work for a default in the other exchanges as well.

Reply:
Their response example on request for getting active order (https://github.com/bybit-exchange/bybit-official-api-docs/blob/master/en/rest_api.md#response-example-5) pointed that order status can be "Filled" which means closed i guess, so for active orders method fetchOrder would be valid. But response of getting conditional orders has another status set considering the example of response from the docs (https://github.com/bybit-exchange/bybit-official-api-docs/blob/master/en/rest_api.md#response-example-12) and seems there is no status that can be defined as closed. I think for now the most optimal decision would be to leave fetchOrder method cause it's the only method that allows to definitely separate orders regarding their type (i mean active or conditional) and probably can return order regardless the state of it.